src/pallene/coder.lua

```diff
@@ -245,17 +245,23 @@ function Coder:get_stack_slot(typ, dst, slot, loc, description_fmt, ...)
     else
         assert(not typ.is_upvalue_box)
         local extra_args = table.pack(...)
+        local expected_type = pallene_type_tag(typ)
+        if expected_type == "LUA_TUSERDATA" then
+            expected_type = C.string(typ.name)
+        else
+            expected_type = "pallene_tag_name(" .. expected_type .. ")"
```

Comment:
Do we still need `pallene_tag_name`, now that we have `pallene_type_name`?

Reply:
We would require both `pallene_tag_name` and `pallene_type_name`. They are different in two ways.
1. Difference in the function signature `const char *pallene_tag_name(int raw_tag)` and `const char *pallene_type_name(lua_State *L, const TValue *v)`
2. `expected_type` is known at the compile time (and uses `pallene_tag_name`) i.e. the argument's expected type. Whereas the argument `const TValue *v` passed to `pallene_type_name` cannot be known at compile time. `v` is the type of the actual argument passed at runtime.

Reply:
1. IIRC, there aren't any other places that call `pallene_tag_name`. The tag name function was there only for the error messages and `pallene_type_name` now does that job. Notice how both functions are currently quite similar.

2. If your goal was that the expected_type be known at compile time, then calling `pallene_tag_name` is not the way to do that. That C function call will only happen at run-time.

3. One last thing: In the coder.lua, please use different variable names for the type tag returned by `pallene_type_tag` and for the type name that you compute inside the if statement.

Reply:
Please have a look at the recent changes. Hope the latest commit resolves the issues/suggestions you mentioned.